diff --git a/ott/mucore.ott b/ott/mucore.ott
index d6371fd..5a2330b 100644
--- a/ott/mucore.ott
+++ b/ott/mucore.ott
@@ -294,7 +294,7 @@ grammar
     | subs ( pvals )        :: M :: Subspvals {{ ocaml SUBS_PVALS [[subs]] [[pvals]] }}

   subrules
-    name <:: pval
+    name <:: pval_aux

 grammar

@@ -311,10 +311,13 @@ grammar

 grammar

-  pat :: 'M_Case' ::= {{ com computational patterns }}
+  pat_aux :: 'M_Case' ::= {{ com computational patterns }}
     | ty_sym_opt_T_bt                   :: :: Base (+ binders = binders( ty_sym_opt_T_bt ) +)
     | ctor_val ( </ pati // , // i /> ) :: :: Ctor (+ binders = binders( </ pati // i /> ) +)

+  pat :: 'M_Case' ::= {{ com because Ott binding doesn't work with aux }}
+    | ty_loc annots pat_aux :: :: Pat_aux (+ binders = binders( pat_aux ) +)
+
 grammar

   sym_or_pat {{ tex ident\_or\_pat }} :: 'M_' ::= {{ auxparam 'TY }} {{ com identifier or pattern }}
